src/lib.rs

```diff
@@ -482,13 +555,13 @@ impl Log for Logger {
                 let mut tl_buf = tl_buf.borrow_mut();
 
                 if tl_buf.is_none() {
-                    *tl_buf = Some(Formatter::new(self.writer.buffer()));
+                    *tl_buf = Some(Formatter::new(self.writer.buffer(), self.write_style));
```

Comment:
This is problematic because multiple loggers working on the same thread with different configurations will get unexpected results. The problem is the same with the `Buffer`, since it's tied to the `Logger`.

I'm thinking we just prevent you building a `Logger` directly and require it's configured globally, but am open to other suggestions.